File: C/Algorithms/2185.c
int prefixCount(char ** words, int wordsSize, char * pref){
    int count = 0;
    for (int i = 0; i < wordsSize; i++) {
        char *p = pref, *word = words[i];
        while (*p++ == *word++){
            if (*p == '\0'){
                count++;
                break;
            }
        }
    }
    return count;
}

File: Rust/Algorithms/2185.rs
impl Solution {
    pub fn prefix_count(words: Vec<String>, pref: String) -> i32 {
        let pref = pref.as_str();
        words
            .iter()
            .filter(|w| w.starts_with(pref))
            .collect::<Vec<_>>()
            .len() as i32
    }
}
